document-portal/file-transfer.c

```diff
@@ -294,11 +298,12 @@ file_transfer_execute (FileTransfer *transfer,
           return NULL;
         }
 
+      the_flags[i] = flags | (file->is_dir ? DOCUMENT_ADD_FLAGS_DIRECTORY : 0);
```

Comment:
Here, I think it would be slightly better to rename the `flags` variable to `common_flags`, and name the array `documents_flags`